### PR TITLE
Adjust VariantQuery to dynamically account for recent last_comment_at

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adjusts existing logic for new behavior — since 18 hours was already hardcoded, I think it's reasonable to inline these numbers — however, could make them configurable later.

## Related Tickets & Documents

- Closes https://github.com/forem/forem/issues/20322

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

File: app/services/articles/feeds/variant_query.rb [hunk 1]
```diff
@@ -261,14 +265,13 @@ def sql_sub_query(limit:, offset:, omit_article_ids:, only_featured: false, must
...
 
       def build_sql_with_where_clauses(only_featured:, must_have_main_image:, omit_article_ids:)
-        where_clauses = "articles.published = true AND articles.published_at > :oldest_published_at"
-        # See Articles.published scope discussion regarding the query planner
+        # Hardcode the values for lookback_hours and comment_hours.
+        where_clauses = "articles.published = true"
         where_clauses += " AND articles.published_at < :now"
-        where_clauses += " AND articles.score >= 0" # We only want positive values here.
-
-        # Without the compact, if we have `omit_article_ids: [nil]` we
```
Comment (lightalloy) [inline app/services/articles/feeds/variant_query.rb:269]: It looks like we still have `omit_article_ids`, should we keep the comment?

File: spec/services/articles/feeds/variant_query_spec.rb [hunk 2]
```diff
@@ -34,6 +34,30 @@
           expect(query_call).to be_a(ActiveRecord::Relation)
           expect(query_call.to_a).not_to match_array(article)
         end
+
+        it "does not return an article published 27 hours before last page view if last comment at is too old" do
+          create_list(:page_view, 5, user: user) # Recent pageviews
```
Comment (lightalloy) [inline spec/services/articles/feeds/variant_query_spec.rb:39]: I would probably DRY the `create_list` and `create(:article)`, but that's just nitpicking.

**Review by lightalloy — APPROVED** [review]
Looks good, I left a couple of small comments.

**Comment by benhalpern** [comment]
@lightalloy Thanks for the comments — I'm going to keep this in mind for some follow-up tweaks

**Pull request merged**